src/SumoCoders/FrameworkCoreBundle/BreadCrumb/BreadCrumbBuilder.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\BreadCrumb;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\MenuItem;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+use SumoCoders\FrameworkCoreBundle\Event\ConfigureMenuEvent;
+
+final class BreadCrumbBuilder
+{
+    /**
+     * @var bool
+     */
+    private $dontExtractFromTheRoute = false;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $factory;
+
+    /**
+     * @var array
+     */
+    private $items = array();
+
+    /**
+     * @param FactoryInterface         $factory
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(FactoryInterface $factory, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->factory = $factory;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Add items into the breadcrumb based on a given child.
+     *
+     * @param MenuItem $item
+     * @param string   $locale
+     */
+    protected function addItemsBasedOnTheChild(MenuItem $item, $locale)
```

Comment:
Ik zou die methods wat herordenen dat de public methods bovenaan staan en de private/protected onderaan staan. De public methods zijn namelijk de methods die je quasi altijd zult nodig hebben.

Daarnaast heeft het niet veel nut om in een final class  methods protected te maken. Ik zou die hier allemaal private maken. (wat trouwens een goeie default is voor alle methods).